### PR TITLE
Interface inhancements, name validation, some fixes

### LINKED ISSUE #58: Change order of elements of interface
1. At lesson's level: move "add step" button to the top of the page;
2. At lesson's level: refactor "make draggable function". An element should be moved by dragging special area (for example marked zone at the end of the line);
3. At course's level: lesson level enchancements + fix random order of courses.

### LINKED ISSUE #73: Validation of steps' and lessons' names
There are some troubles when user creates lesson or step with name which already exists. Also need to check that step/lesson folder's name isn't duplicate of some name of existing file.

### PR DESCRIPTION
**Pull request by pospelov14203**
Resolves #73:
Added name checking (checks that name is unique and OS doesn't contain directory with the same name).

Resolves #58:

- "Add" buttons moved to the top of the page;
- Added list of user's courses to welcome page;
- Delete button for lesson changed to similar to buttons for steps and substeps (line with name and control buttons now contains delete button);
- Special area for dragging elements of the list of step/lesson instead of "MakeDraggable" checkbox. It is much more convenient: no need to click on "MakeDraggable" before drag.
- List of courses shows in creation date order;
- View of courses lists is the same as view of steps/lessons lists.
- Some shaping of UI elements (sizes, margins);
- Removed useless dates in courses names;
- Now navigation elements are in the same style (references to course/lesson);
- Added back buttons wherever they are missing.

Other enhancements:
-Remove error alert when tries to stop non existing FFmpeg process - in such cases, continue to work in normal mode;
-When we tries to stop recording and tablet's FFmpeg termination time is over - anyway tries to download recording (in previous version we didn't try to download if the time is over and  if FFmpeg is still alive).

### CODE REVIEW

File: static/css/default.css [hunk 1]
```diff
@@ -88,13 +91,29 @@
         }
 
         .step_name {
+            word-wrap: break-word;
+            display:inline-block;
+            overflow:hidden !important;
```
Comment (vyahhi) [inline static/css/default.css]: missing space after `:`

File: stepicstudio/templates/lesson_view.html [hunk 2]
```diff
@@ -2,46 +2,48 @@
...
-
-
-        <!--ADD BACK BUTTON -->
-        <br xmlns="http://www.w3.org/1999/html">this page is for Lesson <a href = {{ Lesson.url }}>{{ Lesson.name}}:</a>
+        <b>Course: <a class="reference_wo_underline" href = {{ Course.url }}>{{ Course.name}}</a></b>
         <br>
-        <h3>
-        from  <a href = {{ Course.url }}>{{ Course.name}}</a>
-        </h3>
+        <b>Lesson: <a class="reference_wo_underline" href href = {{ Lesson.url }}>{{ Lesson.name}}</a></b>
```
Comment (vyahhi) [inline stepicstudio/templates/lesson_view.html]: double `href href` not needed here

Comment (vyahhi) [reply stepicstudio/templates/lesson_view.html]: also use quotes for all HTML attrs: `href="{{ Lesson.url }}"`

File: stepicstudio/templates/lesson_view.html [hunk 3]
```diff
@@ -2,46 +2,48 @@
...
 
-        Add new steps here
-
     {% endblock %}
     {% block controls %}
-        <a href="/course/{{ Course.id}}/lesson/{{ Lesson.id  }}/add_step/"> Add Step</a>
-        <div class="delete_button right">
-            <div class="delete-url" data-urllink="/course/{{ Course.id}}/lesson/{{ Lesson.id  }}/delete/">Delete lesson </div>
+        <div class="back button right">
+            <a href= {{Course.url}} >Back</a>
```
Comment (vyahhi) [inline stepicstudio/templates/lesson_view.html]: no need for spaces around `{{Course.url}}`, but there's need for "quotes" around it

File: stepicstudio/templates/lesson_view.html [hunk 4]
```diff
@@ -2,46 +2,48 @@
...
-            <form>
-                Make Draggable <input id ="isDraggable" type="checkbox" value="Draggable">
-            </form>
+        <br>
+        <div style="height: 70px">
+            <div class="add_container">
+                <a class="add_button_ref" href="/course/{{ Course.id}}/lesson/{{ Lesson.id  }}/add_step/"> Add Step</a>
+            </div>
+
+            <div  id="draggable_tip">
```
Comment (vyahhi) [inline stepicstudio/templates/lesson_view.html]: unnesessary extra space between `div` and `id`
(also in few other places in this PR)

File: stepicstudio/views.py [hunk 5]
```diff
@@ -132,11 +131,12 @@ def auth_view(request):
...
         say_hello = bool(request.GET.get('message'))
-        return render_to_response('loggedin.html', {'full_name': request.user.username,
-                                                    'Courses': Course.objects,
-                                                    'say_hello': say_hello,
-                                                    }
-                                  , context_instance=RequestContext(request))
+        args = {'full_name': request.user.username,
+                'say_hello': say_hello,
+                'Courses': Course.objects.filter(editors=request.user.id).order_by('-start_date'),
+                }
```
Comment (vyahhi) [inline stepicstudio/views.py]: Strange `{}` positioning here (also in few other places in this PR). You either put curly braces on separate lines (therefore need a newline after `{`) or put them on the same lines (therefore no need for a newline before `}`).

File: static/css/default.css (hunk 1 above)
Comment (pospelov14203) [reply static/css/default.css]: > missing space after `:`

Fixed in https://github.com/pospelov14203/Stepik-Studio/commit/ded9d3cb56550517b6b09363dab63d15a53a8106.

File: stepicstudio/templates/lesson_view.html (hunk 2 above)
Comment (pospelov14203) [reply stepicstudio/templates/lesson_view.html]: > double `href href` not needed here



> also use quotes for all HTML attrs: `href="{{ Lesson.url }}"`

Fixed in https://github.com/pospelov14203/Stepik-Studio/commit/ded9d3cb56550517b6b09363dab63d15a53a8106.

File: stepicstudio/templates/lesson_view.html (hunk 3 above)
Comment (pospelov14203) [reply stepicstudio/templates/lesson_view.html]: > no need for spaces around `{{Course.url}}`, but there's need for "quotes" around it

Fixed in https://github.com/pospelov14203/Stepik-Studio/commit/ded9d3cb56550517b6b09363dab63d15a53a8106.

File: stepicstudio/templates/lesson_view.html (hunk 4 above)
Comment (pospelov14203) [reply stepicstudio/templates/lesson_view.html]: > unnesessary extra space between `div` and `id`
> (also in few other places in this PR)

Fixed here and in other places in https://github.com/pospelov14203/Stepik-Studio/commit/ded9d3cb56550517b6b09363dab63d15a53a8106.

File: stepicstudio/views.py (hunk 5 above)
Comment (pospelov14203) [reply stepicstudio/views.py]: > Strange `{}` positioning here (also in few other places in this PR). You either put curly braces on separate lines (therefore need a newline after `{`) or put them on the same lines (therefore no need for a newline before `}`).

Fixe here and in other places in https://github.com/pospelov14203/Stepik-Studio/commit/ded9d3cb56550517b6b09363dab63d15a53a8106.

**Pull request merged**